### Binary classification of diabetic patient

# importing and loading of dataset

In [8]:
import numpy as np
import pandas as pd
import os
function=pd.read_csv("diabetes.csv", header=None)
function

0        1              2              3        4     5  \
0    Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI   
1              6      148             72             35        0  33.6   
2              1       85             66             29        0  26.6   
3              8      183             64              0        0  23.3   
4              1       89             66             23       94  28.1   
..           ...      ...            ...            ...      ...   ...   
764           10      101             76             48      180  32.9   
765            2      122             70             27        0  36.8   
766            5      121             72             23      112  26.2   
767            1      126             60              0        0  30.1   
768            1       93             70             31        0  30.4   

                            6    7        8  
0    DiabetesPedigreeFunction  Age  Outcome  
1                       0.627   50        1  
2                       0.351   31        0  
3                       0.672   32        1  
4                       0.167   21        0  
..                        ...  ...      ...  
764                     0.171   63        0  
765                      0.34   27        0  
766                     0.245   30        0  
767                     0.349   47        1  
768                     0.315   23        0  

[769 rows x 9 columns]

# Printing the top head values of dataset

In [9]:
function.head(7)

0        1              2              3        4     5  \
0  Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI   
1            6      148             72             35        0  33.6   
2            1       85             66             29        0  26.6   
3            8      183             64              0        0  23.3   
4            1       89             66             23       94  28.1   
5            0      137             40             35      168  43.1   
6            5      116             74              0        0  25.6   

                          6    7        8  
0  DiabetesPedigreeFunction  Age  Outcome  
1                     0.627   50        1  
2                     0.351   31        0  
3                     0.672   32        1  
4                     0.167   21        0  
5                     2.288   33        1  
6                     0.201   30        0

# Defining the shape of data

In [10]:
function.shape

(769, 9)

# the data set have 769 rows and 9 columns in which 8 columns are attributes and 1 column is outcome

# Defining the size of data

In [11]:
function.size

6921

# Definig all information of data set

In [12]:
function.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 769 entries, 0 to 768
Data columns (total 9 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   0       769 non-null    object
 1   1       769 non-null    object
 2   2       769 non-null    object
 3   3       769 non-null    object
 4   4       769 non-null    object
 5   5       769 non-null    object
 6   6       769 non-null    object
 7   7       769 non-null    object
 8   8       769 non-null    object
dtypes: object(9)
memory usage: 54.2+ KB


# counting the total number of attributes in a data set

In [13]:
function.count()

0    769
1    769
2    769
3    769
4    769
5    769
6    769
7    769
8    769
dtype: int64

# defining the datatypes of each attributes in a dataset

In [14]:
function.dtypes

0    object
1    object
2    object
3    object
4    object
5    object
6    object
7    object
8    object
dtype: object

In [15]:
function.axes

[RangeIndex(start=0, stop=769, step=1),
 Int64Index([0, 1, 2, 3, 4, 5, 6, 7, 8], dtype='int64')]

# statistic representation of dataset with all attributes

In [17]:
function.describe()

0    1    2    3    4    5      6    7    8
count   769  769  769  769  769  769    769  769  769
unique   18  137   48   52  187  249    518   53    3
top       1   99   70    0    0   32  0.254   22    0
freq    135   17   57  227  374   13      6   72  500

# checking missing values of dataset

In [18]:
function.isnull()

0      1      2      3      4      5      6      7      8
0    False  False  False  False  False  False  False  False  False
1    False  False  False  False  False  False  False  False  False
2    False  False  False  False  False  False  False  False  False
3    False  False  False  False  False  False  False  False  False
4    False  False  False  False  False  False  False  False  False
..     ...    ...    ...    ...    ...    ...    ...    ...    ...
764  False  False  False  False  False  False  False  False  False
765  False  False  False  False  False  False  False  False  False
766  False  False  False  False  False  False  False  False  False
767  False  False  False  False  False  False  False  False  False
768  False  False  False  False  False  False  False  False  False

[769 rows x 9 columns]

# checking missing values of dataset in each column

In [19]:
function.isnull().sum()

0    0
1    0
2    0
3    0
4    0
5    0
6    0
7    0
8    0
dtype: int64

# data set do not have any missing value in any of its column so no need to remove any row

### Data set formatting


# By changing the value of any row or column we can format our dataset for making data type compatible with others data type

# formatting of column names

In [20]:
col_name=list('123456789')
function=pd.read_csv(r'diabetes.csv',names=col_name)
function

1        2              3              4        5     6  \
0    Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI   
1              6      148             72             35        0  33.6   
2              1       85             66             29        0  26.6   
3              8      183             64              0        0  23.3   
4              1       89             66             23       94  28.1   
..           ...      ...            ...            ...      ...   ...   
764           10      101             76             48      180  32.9   
765            2      122             70             27        0  36.8   
766            5      121             72             23      112  26.2   
767            1      126             60              0        0  30.1   
768            1       93             70             31        0  30.4   

                            7    8        9  
0    DiabetesPedigreeFunction  Age  Outcome  
1                       0.627   50        1  
2                       0.351   31        0  
3                       0.672   32        1  
4                       0.167   21        0  
..                        ...  ...      ...  
764                     0.171   63        0  
765                      0.34   27        0  
766                     0.245   30        0  
767                     0.349   47        1  
768                     0.315   23        0  

[769 rows x 9 columns]